tcms/core/tests/test_markdown.py

```diff
@@ -0,0 +1,105 @@
+import unittest
+import bleach
+
+from tcms.core.templatetags.extra_filters import markdown2html
+
+
+class TestMarkdownExtraFilters(unittest.TestCase):
+
+    def test_markdown2html_convert_paragraphs(self):
+        self.assertEqual(markdown2html("__*hello!*__"),
+                         ("<p><strong><em>hello!</em></strong></p>"))
+
+    def test_markdown2html_convert_tables(self):
+        self.assertEqual(markdown2html("""|Stx |Desc |\n|----|-----|\n|Head|Title|\n|Text|Text|"""),
+                         ("""<table>
+<thead>
+<tr>
+<th>Stx</th>
+<th>Desc</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td>Head</td>
+<td>Title</td>
+</tr>
+<tr>
+<td>Text</td>
+<td>Text</td>
+</tr>
+</tbody>
+</table>"""))
+
+    def test_markdown2html_convert_nl2br(self):
+        self.assertEqual(markdown2html("""Line 1
+Line 2"""), ("""<p>Line 1<br>
+Line 2</p>"""))
+
+    def test_markdown2html_convert_fenced_code(self):
+        self.assertEqual(markdown2html("""```{
+"firstName": "John",
+"lastName": "Smith",
+"age": 25}``` """), ("""<p><code>{
+"firstName": "John",
+"lastName": "Smith",
+"age": 25}</code> </p>"""))
+
+    def test_markdown2html_bleach_script(self):
+        self.assertEqual(bleach.clean("<b>bold</b> <script>alert('hello')</script>"),
+                         ("<b>bold</b> &lt;script&gt;alert('hello')&lt;/script&gt;"))

```

Comment:
All of the `bleach` scenarios are wrong. SUT is not the bleach.clean() funtion but the markdown2html() function.  After the conversion from Markdown to HTML it uses bleach to clean unwanted HTML tags and has an explicit list of allowed tags & attributes.